scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+		}
+	}
+	public listPaths(): Path[] {
+		if (this.config.paths) {
+			let paths = this.config.paths.map((x) => x.path);
+			return paths;
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	public async getDefaultAccount([chainID]: string[]): Promise<Account> {
+		const chain = this.chainById(chainID);
+		if (chain) {
+			let client = await this.getIBCClient(chain);
+			return {
+				address: client.senderAddress,
+			};
+		} else {
+			throw new Error("Chain not found: " + chainID);
+		}
+	}
+	public async getDefaultAccountBalance([chainID]: string[]): Promise<Coin[]> {
+		const chain = this.chainById(chainID);
+		if (chain) {
+			let client = await this.getIBCClient(chain);
+			return await client.query.bank.allBalances(client.senderAddress);
+		} else {
+			throw new Error("Chain not found: " + chainID);
+		}
+	}
+	public async link(paths: string[]): Promise<LinkResponse> {
+		if (this.config.paths) {
+			let response: LinkResponse = {
+				linkedPaths: [],
+				alreadyLinkedPaths: [],
+			};
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					response.alreadyLinkedPaths.push(pathName);
+				} else {
+					try {
+						await this.createLink(path);
+						response.linkedPaths.push(pathName);
+					} catch (e) {
+						throw new Error("Could not link path: " + pathName + ": " + e);
+					}
+				}
+			}
+			return response;
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	public async start(paths: string[]): Promise<StartResponse> {
+		if (this.config.paths) {
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					const link = await this.getLink(path);
+					this.relayers.set(
+						pathName,
+						setInterval(async () => {
+							let heights = this.pathById(pathName).relayerData;
+							let newHeights = await this.relayPackets(link, heights);
+							this.pathById(pathName).relayerData = newHeights;
+						}, 5000)
+					);
+				} else {
+					throw new Error("Path: " + pathName + " is not linked.");
+				}
+			}
+			return {};
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	// Helper functions
+	protected chainById(chainID: string): ChainConfig {
+		return this.config.chains
+			? this.config.chains.find((x) => x.chainId == chainID)
+			: null;
+	}
+
+	protected pathById(pathID: string): PathConfig {
+		return this.config.paths
+			? this.config.paths.find((x) => x.path.id == pathID)
+			: null;
+	}
+	protected async balanceCheck(chain: ChainConfig): Promise<boolean> {
+		let chainBalances = await this.getDefaultAccountBalance([chain.chainId]);
+		let chainGP = GasPrice.fromString(chain.gasPrice);
+		if (!chainBalances.find((x) => x.denom == chainGP.denom)) {
+			return false;
+		}
+		if (
+			chainBalances.find(
+				(x) =>
+					x.denom == chainGP.denom &&
+					parseInt(x.amount) < chainGP.amount.toFloatApproximation() * 2256000
+			)
+		) {
+			return false;
```

Comment:
no need to have an if statement here, we can directly return !condition

Reply:
ok...